src/consumption.cpp

```diff
@@ -381,7 +381,7 @@ edible_rating player::can_eat( const item &food, bool interactive, bool force )
         overfull = !maybe_query( _( "You're full.  Force yourself to eat?" ) );
     } else if( ( ( nutr > 0 && temp_hunger < capacity ) ||
                  ( comest->quench > 0 && temp_thirst < capacity ) ) &&
-               !eathealth && !slimespawner ) {
+               !food.has_infinite_charges() && !eathealth && !slimespawner ) {
```

Comment:
It prevents the redundant query below. It's obvious that the character won't be able to drink up an infinite source of water. If they have chosen the "Consume it" in the menu, they mean it.